### ISSUE TITLE
ImportError for mxnet: cannot import name libinfo

### ISSUE THREAD

**Issue by gisoo2017** [apache/incubator-mxnet#4746]
## Environment info
Operating System: windows 7, 64bit
MXNet version:7.0 or 5.0 ( tried several different versions for cpu from https://github.com/dmlc/mxnet/releases
Python version and distribution: WinPython-64bit-2.7.10.3

At first when I installed the mxnet, it worked and the second time I tried it gave the following error and I have tried to delete and install the mxnet package several times and it keeps giving the following error :
## Error Message:
import mxnet
---------------------------------------------------------------------------
WindowsError                              Traceback (most recent call last)
<ipython-input-1-30eb4f951ea5> in <module>()
----> 1 import mxnet

C:\Localdata\winpython\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\site-packages\mxnet-0.7.0-py2.7.egg\mxnet\__init__.py in <module>()
      5 
      6 from .context import Context, current_context, cpu, gpu
----> 7 from .base import MXNetError
      8 from . import base
      9 from . import ndarray

C:\Localdata\winpython\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\site-packages\mxnet-0.7.0-py2.7.egg\mxnet\base.py in <module>()
     41 __version__ = libinfo.__version__
     42 # library instance of mxnet
---> 43 _LIB = _load_lib()
     44 
     45 # type definitions

C:\Localdata\winpython\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\site-packages\mxnet-0.7.0-py2.7.egg\mxnet\base.py in _load_lib()
     33     """Load libary by searching possible path."""
     34     lib_path = libinfo.find_lib_path()
---> 35     lib = ctypes.cdll.LoadLibrary(lib_path[0])
     36     # DMatrix functions
     37     lib.MXGetLastError.restype = ctypes.c_char_p

C:\Localdata\winpython\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\ctypes\__init__.pyc in LoadLibrary(self, name)
    441 
    442     def LoadLibrary(self, name):
--> 443         return self._dlltype(name)
    444 
    445 cdll = LibraryLoader(CDLL)

C:\Localdata\winpython\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\ctypes\__init__.pyc in __init__(self, name, mode, handle, use_errno, use_last_error)
    363 
    364         if handle is None:
--> 365             self._handle = _dlopen(self._name, mode)
    366         else:
    367             self._handle = handle

WindowsError: [Error 126] The specified module could not be found

import mxnet
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-2-30eb4f951ea5> in <module>()
----> 1 import mxnet

C:\Localdata\winpython\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\site-packages\mxnet-0.7.0-py2.7.egg\mxnet\__init__.py in <module>()
      5 
      6 from .context import Context, current_context, cpu, gpu
----> 7 from .base import MXNetError
      8 from . import base
      9 from . import ndarray

C:\Localdata\winpython\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\site-packages\mxnet-0.7.0-py2.7.egg\mxnet\base.py in <module>()
      8 import numpy as np
      9 import atexit
---> 10 from . import libinfo
     11 
     12 __all__ = ['MXNetError']

ImportError: cannot import name libinfo

**Comment by szha** [comment]
Hey, thanks for bringing this up. Could you elaborate on the installation process, so that I can try to reproduce the problem? To start, I'm interested in:
1. the exact version of mxnet you tried, and the instructions you followed for installation. just one version is sufficient. a link would be perfect.
2. the link/instructions on getting WinPython-64bit-2.7.10.3

**Comment by gisoo2017** [comment]
1. The first mxnet versions I tried was: windows binary build 20160531_win10_x64_cpu.7z
for installation after I extracted all the files in one folder, I tried to install it using the "WinPython Command Prompt": ...setup.py install. this worked but then I got that error when running in Jupyter notebook.
2. https://sourceforge.net/projects/winpython/files/WinPython_2.7/2.7.10.3/

**Comment by gisoo2017** [comment]
how can I disable cython?

**Comment by QuantumLiu** [comment]
@gisoo2017 Have you solved this problem?I got same problem. #4794

**Comment by gisoo2017** [comment]
yes actually you need to check for the mxnet path in Environment variables of your PC and give the correct path there. In my case it was going wrong there

**Comment by QuantumLiu** [comment]
@gisoo2017 how to set a correct path?

**Comment by gisoo2017** [comment]
just copy and paste the location where you stored the mxnet folder from the window explorer in the Environment variables

**Comment by QuantumLiu** [comment]
i put "C:\Anaconda2\lib\site-packages\mxnet-0.9.3-py2.7.egg\mxnet\"in Environment variables Path,but still not work

**Comment by gisoo2017** [comment]
maybe try to run "setupenv" directly! and then check the path?

**Comment by happygds** [comment]
Did you solve this problem and how to solve it ? @gisoo2017 @QuantumLiu

**Comment by DamonDeng** [comment]
One of the reason for this that you run the python command in the exact directory you install MXNet.

Say that you clone the MXNet project into `/home/user1/incubator-mxnet`, you build the mxnet library then go to the `python` directory to run install command `python setup.py install --user`.

After that , please exit the `python` directory before you run `python` to test mxnet.

As there is a directory named `mxnet` in the `incubator-mxnet/python` directory, if you run `python` in directory `incubator-mxnet/python` and then `import mxnet as mx`, python will try to import the directory instead of the mxnet lib.

In conclusion, do not start python shell in the `incubator-mxnet/python` directory, or else you will get the `cannot import name libinfo`  error message.

Please create a new issue to report it if you didn't run python shell in `incubator-mxnet/python`  while you get the same error message `cannot import name libinfo`

**Comment by szha** [comment]
Shouldn't be a problem any more in the newer versions. Feel free to reopen if issue persists

**Comment by Bumblebee1964** [comment]
Still having this problem.

(C:\Program Files\Anaconda2) C:\Users\Fred>activate mxnetenv

(mxnetenv) C:\Users\Fred>python
Python 2.7.13 |Continuum Analytics, Inc.| (default, May 11 2017, 13:17:26) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import mxnet as mx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\Anaconda2\envs\mxnetenv\lib\site-packages\mxnet-0.11.1-py2.7.egg\mxnet\__init__.py", line 25, in <module>
    from .base import MXNetError
  File "C:\Program Files\Anaconda2\envs\mxnetenv\lib\site-packages\mxnet-0.11.1-py2.7.egg\mxnet\base.py", line 110, in <module>
    _LIB = _load_lib()
  File "C:\Program Files\Anaconda2\envs\mxnetenv\lib\site-packages\mxnet-0.11.1-py2.7.egg\mxnet\base.py", line 102, in _load_lib
    lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_LOCAL)
  File "C:\Program Files\Anaconda2\envs\mxnetenv\lib\ctypes\__init__.py", line 362, in __init__
    self._handle = _dlopen(self._name, mode)
WindowsError: [Error 126] The specified module could not be found
>>>

Used yajidesign release build 'prebuildbase_win10_x64_vc14_v2' and '20170926_mxnet_x64_vc14_gpu' in combination with Anaconda2 / Python 2.7 on Windows 10, 64 bit Professional. As the prebuild comes with cudnn64_7.dll I did not download it as suggested in the readme.txt.

Tried solutions suggested above and in other thread, but getting still stuck with this error message. Package seems to be installed fine in the environment. Checked the PATH variable, as well as tried to set it explicitly in Python before import mxnet.

Any suggestion is welcome.

**Comment by szha** [comment]
@yajiedesign

**Comment by gautamkmr** [comment]
Did we fixed this  ?

**Comment by yajiedesign** [comment]
@Bumblebee1964 do you install cuda 8.0?

**Comment by Bumblebee1964** [comment]
yes I did install cuda 8.0.

I found a workaround based on a post on issue https://github.com/apache/incubator-mxnet/issues/7362 by using: pip install -U --pre mxnet-cu80

**Comment by yajiedesign** [comment]
@Bumblebee1964 ,yes it is simple way,this two way (pip and my rep) use same binary.if you don't want download 200mb+ every time,you can use my rep update it.

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.
